[FE-02] 공통 컴포넌트를 추가하였습니다.

loopy-lim opened the pull request:
## 개요

closed #3 
closed #6 
공통 컴포넌트를 만들었습니다!
만약 추가될 공용 컴포넌트가 있다고 판단되거나, 추가되면 좋을 것 같은 기능이 있으면 말씀해주시면 좋겠습니다.

## 작업사항

### 버튼
- button has isActive
- button은 size와 color을 가집니다.
- size는small, medium, large 3가지를 가집니다.
- color는 primary, secondary, white를 가집니다.

### 아이콘
- icon type is search, back, hamberger, home, close

### 모달창
- modal is composition component
- modal has alert, today modal

### 텍스트
- txt has typography and colors
- txt typography type is h1~h6 and p
- txt colors type is only black

### 패키지 추가
- react-icons
- react-modal

### Br
- 현재 이 브랜치에서 사용할 수 없습니다.

loopy-lim committed:
feat: add common components

- button has isActive

- icon type is search, back, hamberger, home, close

- modal is composition component
- modal has alert, today modal

- button has typography and colors
- button typography type is h1~h6 and p
- button colors type is only black

loopy-lim committed:
feat: add Br

mlnwns commented:
뒤로가기 있으면 좋을거같아요!

mlnwns commented on index.html:
뒤로가기 컴퍼넌트 있으면 좋을거같아요

mlnwns reviewed (commented):
확인했습니다!

mlnwns commented on src/components/common/Br.component.jsx:
수평선 색상이 회색인거같은데, 회원가입 및 학교 인증 페이지에서는 수평선 색상이 대부분 초록색이라 새로 컴포넌트를 만들어야 할까요?

mlnwns replied:
Button 처럼 isActive 통해서 Boolean으로 선택 할 수 있게 하면 되겠네요!

loopy-lim replied:
아마 회원 가입 인증 페이지에서 사용하는 수평선은 input에서 만드는게 좋을꺼 같아요!!
저 수평선은 Br 인 느낌으로 전체적으로 화면을 가를 때 쓰는 요소에요!!

mlnwns commented on src/components/common/Button.component.jsx:
대여내역 페이지의 버튼은 흰색 배경에 초록색 글씨인데 이 부분에 대한 내용도 추가 되어야할까요?

loopy-lim replied:
오!! 그 내역에 대해서 추가할게요! 놓친 부분인거 같아요

loopy-lim committed:
feat: add br has navigate (this bransh can't used)

loopy-lim committed:
feat: add Button size and color type

- add button size small, medium, large
- add button color primary, secondary, white

minkonging commented:
뭔가 보로미 색 #62AB05을 BMColor로 변수명을 지정해서 썼으면 좋겠다는 생각을 했었는데,
막상 피그마와 코드를 보니 버튼에서만 사용해서 딱히 필요 없을 것 같네요
지금 당장은 생각나는게 없고, 어느정도 틀이 잘 되어 있는 것 같은데
페이머니 페이지 만들다 생각나는게 있으면 수시로 FE방에 말씀드리겠습니다!
수고하셨습니다!
